### ISSUE TITLE
Upgrade the Gradle version in the project template

### ISSUE THREAD

**Issue by blasten** [flutter/flutter#49438]
Gradle 6.0 came out recently: https://docs.gradle.org/6.0/release-notes.html

The Flutter template currently uses 5.6. However, 6.0 might become the default version for new Android apps generated by Android Studio.

For context, this is already the case: https://github.com/flutter/flutter/pull/46260#issuecomment-562802455 if a Flutter app is edited on Android Studio.

**Comment by jmagman** [comment]
From https://github.com/flutter/flutter/issues/51464:
> Android Studio 3.6 was released recently, and it seems it now requires a higher version of Gradle than the one Flutter is using.
> 
> ## Steps to Reproduce
> 1. Run `flutter create bug`.
> 2. Update the files as follows: `android/build.gradle` -> `classpath 'com.android.tools.build:gradle:3.6.0'`
> 3. Try to run or build the app for Android.
> 4. The exception provided in the Logs section will appear.
> 
> Flutter projects are currently created using Gradle version `5.4.1`, but to make it compatible with newer versions of Android Studio, it could be bumped to, at least, `5.6.4`, or even `6.2.1` ([latest](https://services.gradle.org/distributions/)).

**Comment by renanyoy** [comment]
trying to build our app from vscode
it was working before
seems related to flutter using an obsolete gradle version 
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> Android resource linking failed
  /Users/renanyoy/git/a2ds/apps/Flutter/leads2keys/build/app/intermediates/merged_manifests/debug/AndroidManifest.xml:22: AAPT: error: unexpected element <queries> found in <manifest>.
```

**Comment by renanyoy** [comment]
fixed with gradle update to 6.7.0 and reverting back build.graddle from 4.0.1 to 3.6.2
```
dependencies {
        classpath 'com.android.tools.build:gradle:3.6.2'
```

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/70808 was reverted.
